### PR TITLE
52349 devops vnet integration

### PR DESCRIPTION
**Pull request by surendrabehara2022**
Hi,
Added IP whitelisting and vnet integration for webapp and keyvault.
Separated mockwebapp app module.

### CODE REVIEW

**Review by mattwin1 — APPROVED** [review]
LGTM

File: DevDeployment/src/Modules/MockWebApp/main.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+resource "azurerm_windows_web_app" "mock_webapp_service" {
+  name                = var.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  service_plan_id     = var.service_plan_id
+  tags                = var.tags
+
+  site_config {
```
Comment (ajasonca) [inline DevDeployment/src/Modules/MockWebApp/main.tf:8]: This section seems to have inconsistent indentation. Please review and format nicely.

File: DevDeployment/src/Modules/MockWebApp/main.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+     dotnet_version = "v6.0"
+    }
+    always_on  = true
+    ftps_state = "Disabled"
+
+   }
+
+  app_settings = var.app_settings
+
+  identity {
```
Comment (ajasonca) [inline DevDeployment/src/Modules/MockWebApp/main.tf:20]: This section also has inconsistent indentation.

File: DevDeployment/src/main.tf [hunk 3]
```diff
@@ -38,7 +51,7 @@ module "mock_webapp_service" {
     "WEBSITE_ENABLE_SYNC_UPDATE_SITE"                      = "true"
     "APPINSIGHTS_INSTRUMENTATIONKEY"                       = "NOT_CONFIGURED"
   }
-  tags = local.tags
+  tags                      = local.tags
```
Comment (ajasonca) [inline DevDeployment/src/main.tf:54]: I think you're trying to line the = sign up with the others, but instead doesn't seem to line up with anything.

File: QALiveDeployment/src/Modules/Webapp/main.tf [hunk 4]
```diff
@@ -13,7 +13,18 @@ resource "azurerm_windows_web_app" "webapp_service" {
     always_on  = true
     ftps_state = "Disabled"
 
+    ip_restriction {
+      virtual_network_subnet_id = var.subnet_id
+    }
+
+    dynamic "ip_restriction" {
+      for_each = var.allowed_ips
+      content {
```
Comment (ajasonca) [inline QALiveDeployment/src/Modules/Webapp/main.tf:22]: The indentation is slightly off here.

**Review by ajasonca — CHANGES_REQUESTED** [review]
Some minor indentation issues

File: DevDeployment/src/Modules/MockWebApp/main.tf (hunk 1 above)
Comment (surendrabehara2022) [reply DevDeployment/src/Modules/MockWebApp/main.tf:8]: Hi,
Fixed the indentation.

File: DevDeployment/src/Modules/MockWebApp/main.tf (hunk 2 above)
Comment (surendrabehara2022) [reply DevDeployment/src/Modules/MockWebApp/main.tf:20]: Hi,
Fixed the indentation.

File: DevDeployment/src/main.tf (hunk 3 above)
Comment (surendrabehara2022) [reply DevDeployment/src/main.tf]: Hi,
Fixed the indentation.

File: QALiveDeployment/src/Modules/Webapp/main.tf (hunk 4 above)
Comment (surendrabehara2022) [reply QALiveDeployment/src/Modules/Webapp/main.tf:22]: Hi,
Fixed the indentation.

File: DevDeployment/src/Modules/MockWebApp/main.tf (hunk 1 above)
Comment (ajasonca) [reply DevDeployment/src/Modules/MockWebApp/main.tf:8]: Lines 9 to 11 are only indented by one space, should be two I think. Also the empty line 15 can be removed.

**Review by ajasonca — CHANGES_REQUESTED** [review]
A remaining indentation issue to fix, otherwise looking good.

File: DevDeployment/src/Modules/MockWebApp/main.tf (hunk 1 above)
Comment (surendrabehara2022) [reply DevDeployment/src/Modules/MockWebApp/main.tf:8]: Fixed the indentation and removed empty line number 15.

**Review by ajasonca — APPROVED** [review]
Looks good, thanks.

**Pull request merged**